src/functional/program_processors/codegens/gtfn/itir_to_gtfn_ir.py

```diff
@@ -149,17 +153,18 @@ def _collect_offset_definitions(
             raise ValueError(f"Missing offset_provider entry for {o.value}")
 
         offset_name = o.value
-        if isinstance(offset_provider[o.value], common.Dimension):
-            dim = offset_provider[o.value]
+        if isinstance(offset_provider[offset_name], common.Dimension):
```

Comment:
```suggestion
        dim_or_conn = offset_provider[offset_name]
        if isinstance(dim_or_conn, common.Dimension):
``` 
then can undo the other changes and don't need ignore